pyzebra/fit2.py

```diff
@@ -0,0 +1,151 @@
+
+        def lorentzian(x, l_cen, l_width, l_amp):
+            """1d lorentzian"""
+            return (l_amp / (1 + ((1 * x - l_cen) / l_width) ** 2)) / (np.pi * l_width)
+
+        def background(x, slope, intercept):
+            """background"""
+            return slope*x + intercept
+
+        mod = Model(gaussian) + Model(lorentzian) + Model(background)
+        params = Parameters()
+        params.add_many(('g_cen', x[int(peak_index)], bool(vary[0]), np.min(x), np.max(x), None, None),
+                        ('g_width', guess[1], bool(vary[1]), constraints_min[1], constraints_max[1], None, None),
+                        ('g_amp', guess[2], bool(vary[2]), constraints_min[2], constraints_max[2], None, None),
+                        ('l_cen', guess[3], bool(vary[3]), np.min(x), np.max(x), None, None),
+                        ('l_width', guess[4], bool(vary[4]), constraints_min[4], constraints_max[4], None, None),
+                        ('l_amp', guess[5], bool(vary[5]), constraints_min[5], constraints_max[5], None, None),
+                        ('slope', guess[6], bool(vary[6]), constraints_min[6], constraints_max[6], None, None),
+                        ('intercept', guess[7], bool(vary[7]), constraints_min[7], constraints_max[7], None, None))
+
+        result = mod.fit(y, params, x=x)
+        print('Chi-sqr: ', result.chisqr)
+
+        comps = result.eval_components()
+
+        gauss_3sigmamin = find_nearest(x, result.params['g_cen'].value-3*result.params['g_width'].value)
+        gauss_3sigmamax = find_nearest(x, result.params['g_cen'].value+3*result.params['g_width'].value)
+        numfit_min = gauss_3sigmamin if numfit_min is None else find_nearest(x, numfit_min)
+        numfit_max = gauss_3sigmamax if numfit_max is None else find_nearest(x, numfit_max)
+        print(numfit_max, numfit_min)
+        if x[numfit_min] < np.min(x):
+            numfit_min = gauss_3sigmamin
+            print('Minimal integration value outside of x range')
+        elif x[numfit_min] >= x[numfit_max]:
+            numfit_min = gauss_3sigmamin
+            print('Minimal integration value higher than maximal')
+        else:
+            pass
+        if x[numfit_max] > np.max(x):
+            numfit_max = gauss_3sigmamax
+            print('Maximal integration value outside of x range')
+        elif x[numfit_max] <= x[numfit_min]:
+            numfit_max = gauss_3sigmamax
+            print('Maximal integration value lower than minimal')
+        else:
+            pass
+        print(result.params['g_width'].value)
+        print(result.params['g_cen'].value)
+        num_int_area = simps(y[numfit_min:numfit_max], x[numfit_min:numfit_max])
+        num_int_bacground = integrate.quad(background, numfit_min, numfit_max, args=(result.params['slope'].value,result.params['intercept'].value))
+
+        plt.plot(x, y, 'b', label='Original data')

```

Comment:
Fitting and plotting needs to be separated into different functions. Otherwise, I won't be able to use only fitting in the web app, since the plotting part will be different.

Reply:
The plotting is not integral part to the function - I used to see what are the results and though it might help you to get the right data without having to search the dictionary. Therefore, it can be deleted or commented

Reply:
I would delete it then. However, it might be useful as a separate utility function in case of using jupyter notebook directly. Feel free to extract it, or remove the lines completely.

Reply:
I removed it, but still have the original file which I use to verification, so if it would be useful or if you would like to call it as s function, let me know.